### PR TITLE
check for conda < 4.3

### PR DESCRIPTION
**Pull request by jcb91**
as discussed in https://github.com/conda-forge/conda-smithy/issues/494
check for conda < 4.3, and if not found, print a message more informative than the exception which would otherwise arise (since currently we're incompatible with 4.3+)

### CODE REVIEW

File: conda_smithy/cli.py [hunk 1]
```diff
@@ -272,6 +273,13 @@ class Rerender(Regenerate):
     else:
         args = parser.parse_args()
 
+    # Check conda version for compatibility
+    conda_version_tuple = tuple(map(int, conda.__version__.split('.')[:3]))
+    if conda_version_tuple > (4, 2):
```
Comment (jakirkham) [inline conda_smithy/cli.py]: `distutils` has some version comparison stuff that might be helpful. Would check that out.

**Comment by jcb91** [comment]
sure, could do something like

```python
from distutils.version import LooseVersion, StrictVersion
if LooseVersion(conda.__version__) < LooseVersion('4.3')
```

?

**Comment by jakirkham** [comment]
Yep, something like that could work.

**Comment by jcb91** [comment]
Ok, done. However, as noted in [the stackoverflow post](http://stackoverflow.com/questions/11887762) where I found that distutils comparison, 

> `distutils.version` is built in but __is undocumented and conformant only to the superseded PEP 386__

(emphasis mine)

**Comment by jakirkham** [comment]
Mainly suggested it because the previous check was failing and this would be an easy fix.

While it is true that they are undocumented and rely on an old superseded PEP, there is a lot of usage of both [`LooseVersion`]( https://github.com/search?utf8=%E2%9C%93&q=distutils+LooseVersion&type=Code ) and [`StrictVersion`]( https://github.com/search?utf8=%E2%9C%93&q=distutils+StrictVersion&type=Code ). In fact, I think the main reason they haven't been fixed up to match PEP440 and documented is because there is fear of breaking existing people's code. I don't think `distutils.version` is going to disappear quickly and quietly. Maybe by Python 4 it will be gone. Hopefully by that time, `conda` does something smarter and we don't need this warning at all. If not, as you have noted, it is easy enough to fix when the time comes.

TL;DR Noted the concern, but am not too worried about it.

**Comment by jcb91** [comment]
Yep, fair enough, I think it's probably sufficient...

File: conda_smithy/cli.py [hunk 2]
```diff
@@ -272,6 +274,12 @@ class Rerender(Regenerate):
     else:
         args = parser.parse_args()
 
+    # Check conda version for compatibility
+    if LooseVersion(conda.__version__) >= LooseVersion('4.3'):
+        print('You appear to be using conda {}, but conda-smithy {} is\ncurrently only compatible with conda versions < 4.3.'.format(
+            conda.__version__, __version__))
+        sys.exit(2)
```
Comment (jakirkham) [inline conda_smithy/cli.py]: Why 2?

**Comment by jcb91** [comment]
Well, I saw 1 being used already (I think for linting errors?), so thought it might be good to distinguish this failure case. But of course, it needn't be 2, if you'd rather keep it simpler!

**Comment by jakirkham** [comment]
Sure that's fine. Just wanted to know if there was some reason.

**Comment by jakirkham** [comment]
Thanks @jcb91.
